pkg/comp-functions/functions/vshnnextcloud/files/install-collabora.sh

```diff
@@ -0,0 +1,37 @@
+ /bin/bash -x
```

Comment:
This script would be more readable if you put all the commands in a script configMap that gets mounted to the pods. Then we just need to do one exec here that calls this script.

Reply:
can You reword what You wrote? maybe some small example?

Reply:
Instead of running this script with a lot of separate `/usr/bin/kubectl exec -n "$NAMESPACE" "deployments/$NEXTCLOUD_INSTANCE" -- ...` commands. You could have the script separately, then mount it into the nextcloud pod, for example in `/scripts/installcollabora.sh`

And here in this script you can just do one line: `/usr/bin/kubectl exec -n "$NAMESPACE" "deployments/$NEXTCLOUD_INSTANCE" -- /scripts/installcollabora.sh`

This should make it a lot more readable.

Reply:
but I'd have to add it in deploy.go which manages bare Nextcloud. I like it better here, so we have everything logically splitted. I'm not saying no to this idea, but as I said, currently collabora logic resides completely in collabora.go and nextcloud in deploy.go.

Reply:
There are two ways to logically split this.

1. If collabora is enabled we adjust the nextcloud values in `deploy.go`. We're already doing similar things for other services
2. We change the values of the nextcloud release in `collabora.go`. This is also something you'll find in our services, especially for the ones were we had PnT and comp functions mixed...

I'd go with the first approach, as it's much cleaner.

Any other thoughts? @zugao

Reply:
We have to understand that since we agreed that we will use one collabora instance per each nextcloud instance we sealed the fate there. Collabora per se cannot pe called a separate VSHN service which means it's just an add-on, like enabling repack on PostgreSQL but much much fatter and more complex to enable. That being said, I don't see why we have to separate them logically? 

{quote}
If collabora is enabled we adjust the nextcloud values in deploy.go
{quote}
This makes a lot of sense. We don't treat collabora at nextcloud level, collabora is just an add-on and we have to treat it as such.

Reply:
but with separation we don't trigger Nextcloud restart when someone enables Collabora, if I start mounting stuff to Nextcloud pods, then it will trigger restart as well as if script changes. In a way it's currently deployed, customer can enable/disable Collabora and users can continue working on Nextcloud without issues

Reply:
Then we just mount the script unconditionally.